tuf/api/metadata.py

```diff
@@ -1399,6 +1536,9 @@ def is_delegated_path(self, target_filepath: str) -> bool:
                 targets URL.
         """
 
+        msg = "is_delegated_path is depricated, use find_delegation instead!"
+        warnings.warn(msg, DeprecationWarning)
```

Comment:
it's not quite this simple if you want to keep using this method in find_delegation() for now: now calling find_delegation(), a non-deprecated function, will result in a deprecation warning.

So either we don't have the warning (this is fine IMO: switching from `is_delegated_path()` to `find_delegation()` does not require any preparation) or we prevent the warning in the non-deprecated code path.